app/services/custom_fields/hierarchy/hierarchical_item_service.rb

```diff
@@ -92,6 +92,7 @@ def delete_branch(item:)
         custom_field = item.root&.custom_field
         if item.destroy
           update_calculated_values_for_hierarchy(item_ids:, custom_field:)
+          remove_assigned_custom_values(custom_field_id: custom_field.id, item_ids:)
```

Comment:
Is this service guaranteed to run these things in a transaction?

Right now I see the potential that the `item.destroy` call succeeds, but then an error occurs when deleting the custom values (e.g. database timeout) or even before, when updating the calculated values.

If not covered by a transaction, this would still leave the database in the invalid state.

:red_circle: **tl;dr:** I'd either make sure that this code always runs in a transaction **or** that it's safe to not run to completion, e.g. by first deleting custom values and then deleting the custom field.

Reply:
Since you now added the transaction, I don't care about the execution order anymore :)